src/base-input.js

```diff
@@ -442,6 +453,14 @@ export default class BaseInput extends LitElement {
     // Prevents cursor jumping in Safari.
     const { selectionStart } = this.inputElement;
 
+    this.inputElement.addEventListener('keyup', (evt) => {
+      if (evt.key === 'Backspace') {
+        if (this.type === 'month-day-year' && this.value) {
+          this.inputElement.setSelectionRange(selectionStart, selectionStart);
+        }
+      }
+    });
+
```

Comment:
This isn't handling when CleaveJS is auto-formatting the input. I have an idea. Let's do some pair-programming today.